universum/modules/vcs/vcs.py

```diff
@@ -87,19 +97,9 @@ def __init__(self, *args, **kwargs):
                 raise IncorrectParameterError(text)
 
             try:
-                if self.settings.type == "none":
-                    driver_factory = self.local_driver_factory
-                elif self.settings.type == "git":
-                    driver_factory = self.git_driver_factory
-                elif self.settings.type == "gerrit":
-                    driver_factory = self.gerrit_driver_factory
-                elif self.settings.type == "github":
-                    driver_factory = self.github_driver_factory
-                else:
-                    driver_factory = self.perforce_driver_factory
-            except AttributeError:
+                self.driver = driver_factory[self.settings.type]()
+            except AttributeError | KeyError:
```

Comment:
I assume KeyError can never happen here due to option filtering. Should I remove this precaution?

Reply:
I'd say, yes.